Fri Jun 23 15:04:41 CEST 2006  VMiklos <vmiklos@frugalware.org>
  * libfwutil: always check for /proc in fwutil_init()
  closes #869
diff -rN -u old-frugalwareutils/libfwutil/libfwutil.c new-frugalwareutils/libfwutil/libfwutil.c
--- old-frugalwareutils/libfwutil/libfwutil.c	2006-06-23 17:36:16.000000000 +0200
+++ new-frugalwareutils/libfwutil/libfwutil.c	2006-06-23 17:36:16.000000000 +0200
@@ -134,17 +134,21 @@
 }
 
 /** Initialize the environment if necessary
- * @return 1 if fwx_release() call is needed later, 0 if not
+ * @return true if fwx_release() call is needed later, false if not
  */
 int fwutil_init()
 {
 	struct stat buf;
 	FILE *fi, *fo;
+	int ret=0;
 
+	if(stat("/proc/1", &buf))
+	{
+		system("mount /proc");
+		ret++;
+	}
 	if(stat("/dev/zero", &buf))
 	{
-		if(stat("/proc/1", &buf))
-			system("mount /proc");
 		system("/etc/rc.d/rc.udev");
 		system("mount / -o rw,remount");
 
@@ -167,9 +171,9 @@
 		}
 
 		system("mount /dev/pts");
-		return(1);
+		ret++;
 	}
-	return(0);
+	return(ret);
 }
 
 /** Release the environment

